src/js/dqa.jsx

```diff
@@ -21,122 +15,212 @@ var SortTypes = {
         
 var DQA = React.createClass({
     propTypes: {
-        content: React.PropTypes.string,
-        width: React.PropTypes.number,
+        aggregated: React.PropTypes.bool,
         data: React.PropTypes.object,
+        dict: React.PropTypes.object,
     },
 
     getDefaultProps: function() {
         return {
-            content: '',
-            width: 500,
+            aggregated: false,
             data: {},
+            dict: {}
         };
     },
 
     getInitialState: function() {
         return {
+            columnWidths: {
+                code: width,
+                desc: 3*width,
+                status: 2*width,
+                sites: 3*width,
+            },
             isCollapsed: false,
-            rows: this._getRows(),
-            sortBy: null,
-            sortDir: null,
+            rows: {},
+            sortBy: 'code',
+            sortDir: SortTypes.ASC,
+            tableWidth: 500,
         };
     },
 
-    _getRows: function() {
+    _getRows: function(props, sortBy) {
         var code;
+        var desc;
         var rows = [];
         var status;
-        var numSites;
+        var sites;
+
+        if (!sortBy) {
+            sortBy = this.state.sortBy;
+        }
+
+        for (code in props.data) {
+            desc = props.dict[code]? props.dict[code].desc : '';
+
+            for (status in props.data[code]) {
+                if (this.props.aggregated) {
+                    sites = [];
+                    for (site in props.data[code][status]) {
+                        sites.push(site + '(' + props.data[code][status][site].length + ')');
+                    }
+                } else {
+                    sites = props.data[code][status].map(function(obj) {
+                        return obj.site;
+                    });
+                }
 
-        for(code in this.props.data) {
-            for(status in this.props.data[code]) {
-                numSites = this.props.data[code][status].length;
-                rows.push([code, '', status, numSites]);
+                // if the table is sorted by site, list sites individually;
+                // if the table is sorted by anything other than site, 
+                // keep sites grouped by (code + status)
+                if (sortBy === 'sites') {
+                    for (i in sites) {
+                        rows.push({
+                            code: code,
+                            desc: desc,
+                            status: status,
+                            sites: sites[i],
+                        });
+                    }    
+                } else {
+                    sites = sites.reduce(function(a,b) {
+                        return a + ', ' + b;
+                    });
+
+                    rows.push({
+                        code: code,
+                        desc: desc,
+                        status: status,
+                        sites: sites,
```

Comment:
Hah didn't know that, thanks! And here I thought that this was an intentional super-convenient feature, so that you can add and remove rows without worrying which line needs a comma and which doesn't. I'll clean them up.